src/posts/blog/3-hepa-filters/HepaCalculator.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from 'react'
+import { Col, Row } from 'react-bootstrap'
+
+interface Dimensions {
+  length: number
+  width: number
+  height: number
+  unit: string
+}
+
+const DimensionControl: React.FunctionComponent<{
+  label: string
+  value: number
+  setter: (val: number) => void
+  unit?: string
+}> = ({ label, value, setter, unit = 'ft' }) => {
+  return (
+    <>
+      <label>
+        <strong>
+          {label} ({unit})
+        </strong>
+        :
+        <input
+          className="form-control form-control-lg col-md-6"
+          type="number"
+          value={value}
+          onChange={(e) => setter(Math.max(0, parseFloat(e.target.value)))}
+        />
+      </label>
+    </>
+  )
+}
+
+const FEET_PER_METER = 3.28084
+const CFM_PER_CMH = FEET_PER_METER ** 3 / 60
+
+export const HepaCalculator: React.FunctionComponent = () => {
+  const [dimensions, setDimensions] = useState<Dimensions>({
+    length: 12,
+    width: 10,
+    height: 8,
+    unit: 'ft',
+  })
+
+  const toggleUnits = () => {
+    let unit
+    let multiplier
+    if (dimensions.unit === 'ft') {
+      unit = 'm'
+      multiplier = 1 / FEET_PER_METER
+    } else {
+      unit = 'ft'
+      multiplier = FEET_PER_METER
+    }
+    setDimensions({
+      unit,
+      // Round to one decimal place
+      length: Math.round(dimensions.length * multiplier * 10) / 10,
+      width: Math.round(dimensions.width * multiplier * 10) / 10,
+      height: Math.round(dimensions.height * multiplier * 10) / 10,
+    })
+  }
+
+  let cfm // cubic feet per minute
+  let cmh // cubic feet per hour (yes, metric folks use a different time base)
+  if (dimensions.unit === 'ft') {
+    cfm = (5 * (dimensions.length * dimensions.width * dimensions.height)) / 60
+    cmh = cfm / CFM_PER_CMH
+  } else {
+    cmh = 5 * (dimensions.length * dimensions.width * dimensions.height)
+    cfm = cmh * CFM_PER_CMH
+  }
+
+  cfm = cfm.toFixed(0)
+  cmh = cmh.toFixed(0)
+
+  return (
+    <div className="card">
+      <header id="filter-calculator">HEPA Filter Calculator</header>
+      <div className="subheading">
+        Enter your room's dimensions to see the minimum flow rate needed to
```

Comment:
Oh one last thing. It'd be nice if the description said something like "the minimum flow rate that your HEPA filter is rated for." 

And if you wanted to go a step further you could say "You can look on the product description page for "flow rate" or "CADR" to see if a HEPA filter is sufficient for your needs."

Reply:
Oh, but that's also wrong, since if you get a hepa filter rated for x CFM and then run it on low, it's not moving enough air.

I've never seen the words "flow rate" on a product page.

Messed with the wording a bit.